File: 1768-merge-strings-alternately/1768-merge-strings-alternately.c
char * mergeAlternately(char * word1, char * word2){
    int l1 = strlen(word1);
    int l2 = strlen(word2);
    int i = 0,j = 0;
    char *merged = malloc(sizeof(char) * (l1+l2) + 1);
    
    while(word1[i] && word2[i]){
        merged[j] = word1[i];
        merged[j+1] = word2[i];
        i++;
        j += 2;
    }
    
    if(word1[i]){
        while(word1[i])
            merged[j++] = word1[i++];
    }
    else if(word2[i]){
        while(word2[i])
            merged[j++] = word2[i++];
    }
    
    merged[j] = '\0';
    return merged;
}

File: 1768-merge-strings-alternately/1768-merge-strings-alternately.swift
class Solution {
    func mergeAlternately(_ word1: String, _ word2: String) -> String {
        var i = 0
        var j = 0
        var arr = [Character]()
        
        while i < word1.count || j < word2.count {
            if i < word1.count {
                let index = word1.index(word1.startIndex, offsetBy: i)
                arr.append(word1[index])
                i += 1
            }
            if j < word2.count {
                let index = word2.index(word2.startIndex, offsetBy: j)
                arr.append(word2[index])
                j += 1
            }
        }
        return String(arr)
    }
}